docs/DEVELOPMENT-TIPS-TRICKS.md

```diff
@@ -0,0 +1,184 @@
+most easy and straightforward task.
+
+In this section of the documentation you will find hints on how to improve your
+development workflows in a way that will make you work faster and easier with
+the codebase locally.
+
+## Developing an ETL service
+
+EUBFR is a data lake project. ETLs are the plugins being used for transforming
+producers' data from one format and structure to a [desired project model target][1].
+
+Normally, when new producers are introduced in the system, they provide raw files
+which serve as the base for building ETLs. Use these raw files for:
+
+- Create a mapping document in the corresponding ETL's folder.
+- Open a pull request to build a preview and reach an agreement about the mapping.
+- Use this mapping agreement as a base for the development of the transform function.
+
+Example: Imagine you receive a file `foo.csv` from `BAR` producer. In this case:
+
+- Create the mapping document in `/services/ingestion/etl/bar/csv/README.md`
+  You can find an example [here][2]
+- Open a pull request, like [this][3]
+- Use this mapping as described in the next section
+
+### Developing a new ETL
+
+Producers' raw data files vary in format and structure, but all of them should
+reach a consistent structure and format. This stage is the "harmonization" stage.
+Transform functions are functions which are called on each record/row of data
+and output of the process is a new line delimited JSON file stored in harmonized
+storage.
+
+This is the conventional structure of an ETL:
+
+```
+.
+└── csv
+    ├── node_modules
+    ├── package.json
+    ├── README.md
+    ├── serverless.yml
+    ├── src
+    │   ├── events
+    │   │   └── onParseCSV.js
+    │   └── lib
+    │       ├── sns.js
+    │       ├── transform.js
+    │       └── uploadFromStream.js
+    ├── test
+    │   ├── stubs
+    │   │   └── record.json
+    │   └── unit
+    │       ├── events
+    │       │   └── onParseCSV.spec.js
+    │       └── lib
+    │           ├── __snapshots__
+    │           │   └── transform.spec.js.snap
+    │           └── transform.spec.js
+    └── webpack.config.js
+```
+
+Here's a possible workflow which has proved efficient so far:
+
+1.  Create the skeleton of the [serverless service][4]
+
+It includes:
+
+- serverless service manifest file: `serverless.yml` - defines cloud resources
+  and glue between logic of the service
+- AWS lambda function which gets triggered when a new file for the ETL comes
+  in - in this case it's `./src/events/onParseCSV.js`. Without going into too
+  many details here - you can take an example from another service matching the
+  file format (extension) from another service. (CSV for example)
+- Creating boilerplate assets: `test` folder, `webpack.config.js` file, etc. These
+  you possibly already have taking the example service from the previous step.
+
+2.  Grab an example record structure
+
+- Deploy the service with `npx serverless deploy` ran inside the root of the service
```

Comment:
`run`, not `ran` 👼

Reply:
It's the past tense in the meaning of "being ran". I'll simplify it though.